src/io/image_io.cc

```diff
@@ -218,29 +218,29 @@ void Imread(const nnvm::NodeAttrs& attrs,
   const auto& param = nnvm::get<ImreadParam>(attrs.parsed);
 
   std::ifstream file(param.filename, std::ios::binary | std::ios::ate);
+  // if file is not open we get bad alloc after tellg
+  CHECK(file.is_open()) << "Imread: Couldn't open file: " << param.filename;
   size_t fsize = file.tellg();
   file.seekg(0, std::ios::beg);
-  auto buff = new uint8_t[fsize];
-  file.read(reinterpret_cast<char*>(buff), fsize);
+  auto buff = std::make_shared<std::vector<uint8_t> >(fsize);
```

Comment:
Why does this need to be shared rather than unique_ptr?  If shared isn;t required, then unique is preferred.

Reply:
because it's passed to the engine below, and can be destroyed asynchronously / from another thread. Can you verify this?

Reply:
It does not need to be shared because it's not used after passing to the PushSync lambda, so it can just be passed by value, the local one being reset to null when losing ownership.